source/modulo_core/include/modulo_core/communication/PublisherHandler.hpp

```diff
@@ -99,6 +108,21 @@ void PublisherHandler<PubT, MsgT>::on_deactivate() {
   }
 }
 
+template<typename PubT, typename MsgT>
+void PublisherHandler<PubT, MsgT>::publish() {
+  if (this->message_pair_ == nullptr) {
+    throw exceptions::NullPointerException("Message pair is not set, cannot deduce message type");
+  }
+  if (this->message_pair_->get_type() == MessageType::CUSTOM_MESSAGE) {
```

Comment:
Ok, indeed this is a bit hacky, but I don't have a cleaner solution yet. We can give it a moment of thought later on in this PR or I can leave a TODO to refactor.

- What it does is:

The problem starts from the fact that for generic types we need to have `MsgT` or explicitly know it. This is ok for translated types, hence `PublisherInterface::publish` can be used (else block).

For generic types we need to
1. override the publish in `PublisherHandler` otherwise we can't deduce `MsgT`
2. and then explicitly call `publish(MsgT&)` which is compatible for all types

- Why the constexpr block

Since both generic and translated types will go through this publish function, it means that the compiler needs to resolve `write<MsgT, MsgT>` for all `MessageType`s even though there is a runtime condition for `MessageType::CUSTOM_MESSAGE`. 

The solution is to introduce a secondary compile-time condition to hide this block when translated types are used. 


I currently don't have a better alternative to this.